x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/textembedding/TextEmbeddingInternalService.java

```diff
@@ -0,0 +1,400 @@
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelInput;
+import org.elasticsearch.xpack.core.ml.inference.results.ChunkedTextExpansionResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextEmbeddingConfigUpdate;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextExpansionConfigUpdate;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TokenizationConfigUpdate;
+import org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.xpack.core.ClientHelper.INFERENCE_ORIGIN;
+import static org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeFromMapOrThrowIfNull;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.throwIfNotEmptyMap;
+import static org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings.MODEL_ID;
+
+public class TextEmbeddingInternalService implements InferenceService {
+
+    public static final String NAME = "text_embedding";
+
+    static final String MULTILINGUAL_E5_SMALL_MODEL_ID = ".multilingual-e5-small";
+    static final String MULTILINGUAL_E5_SMALL_MODEL_ID_LINUX_X86 = ".multilingual-e5-small_linux-x86_64";
+    public static final Set<String> MULTILINGUAL_E5_SMALL_VALID_IDS = Set.of(
+        MULTILINGUAL_E5_SMALL_MODEL_ID,
+        MULTILINGUAL_E5_SMALL_MODEL_ID_LINUX_X86
+    );
+
+    private final OriginSettingClient client;
+
+    private static final Logger logger = LogManager.getLogger(TextEmbeddingInternalService.class);
+
+    public TextEmbeddingInternalService(InferenceServiceExtension.InferenceServiceFactoryContext context) {
+        this.client = new OriginSettingClient(context.client(), ClientHelper.INFERENCE_ORIGIN);
+    }
+
+    @Override
+    public void parseRequestConfig(
+        String inferenceEntityId,
+        TaskType taskType,
+        Map<String, Object> config,
+        Set<String> platformArchitectures,
+        ActionListener<Model> modelListener
+    ) {
+        try {
+            Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
+            String modelId = (String) serviceSettingsMap.get(MODEL_ID);
+            if (modelId == null) {
+                throw new IllegalArgumentException("Error parsing request config, model id is missing");
+            }
+            if (MULTILINGUAL_E5_SMALL_VALID_IDS.contains(modelId)) {
+                e5Case(inferenceEntityId, taskType, config, platformArchitectures, serviceSettingsMap, modelListener);
+            } else {
+                customElandCase(inferenceEntityId, taskType, serviceSettingsMap, modelListener);
+            }
```

Comment:
Should we do the `throwIfNotEmptyMap` calls here to handle both the `e5Case` and `customElandCase`.

Reply:
Also could we add a test for that case?

Reply:
I tried adding that there, but because the response is an actionListener they wouldn't get called in time, so I added them to the respective case functions instead.